### PR TITLE
Broken bulk export

### LINKED ISSUE #4013: Subsequent Bulk Export calls fail
* First bulk export works.
* Second one (identical request) fails to find any patients. 
* Any subsequent request on the same resource type fails similarly.

### PR DESCRIPTION
**Pull request by tadgh**
Closes #4013 

* Remove caches for search builders, as they are not reusable. 
* Remove FHIRPath cache. 
* Add test
* Add changelog

### CODE REVIEW

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/bulk/export/svc/JpaBulkExportProcessor.java [hunk 1]
```diff
@@ -151,8 +147,10 @@ public Iterator<ResourcePersistentId> getResourcePidIterator(ExportPIDIteratorPa
 				}
 			}
 		} else if (theParams.getExportStyle() == BulkDataExportOptions.ExportStyle.GROUP) {
+			ourLog.info("We are about to expand a Group Export");
```
Comment (jamesagnew) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/bulk/export/svc/JpaBulkExportProcessor.java:150]: this line and the other log below should probably be dropped (or set to `trace`)

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/bulk/export/svc/JpaBulkExportProcessor.java [hunk 2]
```diff
@@ -258,6 +249,7 @@ private void validateSearchParametersForGroup(SearchParameterMap expandedSpMap,
 	private Iterator<ResourcePersistentId> getExpandedPatientIterator(ExportPIDIteratorParameters theParameters) {
 		List<String> members = getMembersFromGroupWithFilter(theParameters);
 		List<IIdType> ids = members.stream().map(member -> new IdDt("Patient/" + member)).collect(Collectors.toList());
+		ourLog.info("While extracting patients from a group, we found {} patients.", ids.size());
```
Comment (jamesagnew) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/bulk/export/svc/JpaBulkExportProcessor.java:252]: For what it's worth, i'd leave this log line, as well as the one below, but maybe make them `debug`.

Also recommend leaving the new `setLoadSynchronous` in place. There's no real advantage to doing that fetch asynchronously.

**Review by jamesagnew — APPROVED** [review]

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/bulk/export/svc/JpaBulkExportProcessor.java (hunk 2 above)
Comment (tadgh) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/bulk/export/svc/JpaBulkExportProcessor.java:252]: Roger, fixing now

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/bulk/export/svc/JpaBulkExportProcessor.java (hunk 1 above)
Comment (tadgh) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/bulk/export/svc/JpaBulkExportProcessor.java:150]: done

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/bulk/export/svc/JpaBulkExportProcessor.java (hunk 2 above)
Comment (tadgh) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/bulk/export/svc/JpaBulkExportProcessor.java:252]: actually, turns out the SP Map builder in use here already sets it to load synchronous, so mine is redundant.

File: hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/bulk/BulkDataExportTest.java [hunk 3]
```diff
@@ -90,11 +90,38 @@ public void testGroupBulkExportNotInGroup_DoeNotShowUp() {
...
+			verifyBulkExportResults("G2", new HashSet<>(), List.of("\"PING1\""), Collections.singletonList("\"PNING3\""));
+		} finally {
+			myCaptureQueriesListener.logSelectQueries();
+
+		}
 	}
 
 
-	private void varifyBulkExportResults(String theGroupId, HashSet<String> theFilters, List<String> theContainedList, List<String> theExcludedList) {
+	private void verifyBulkExportResults(String theGroupId, HashSet<String> theFilters, List<String> theContainedList, List<String> theExcludedList) {
```
Comment (StevenXLi) [inline hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/bulk/BulkDataExportTest.java:124]: Thanks for fixing the typo, oops 😅

Comment (tadgh) [reply hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/bulk/BulkDataExportTest.java:124]: 💯

**Pull request merged**